src/g3log/g3log.hpp

```diff
@@ -149,6 +154,14 @@ namespace g3 {
 #define CHECK(boolean_expression)        \
    if (false == (boolean_expression))  INTERNAL_CONTRACT_MESSAGE(#boolean_expression).stream()
 
+#ifdef ENABLE_WIN_WSTRING_SUPPPORT
+#	define LOGW(level) if(g3::logLevel(level)) INTERNAL_LOG_MESSAGE(level).wstream()
```

Comment:
The gain is only in performance. All is based on the assumption that at this point I do not know which encoding the sink will prefer and that I want to log as quickly as possible. Thus when having the choice I would always send an Utf-8 encoded narrow string. However, when I want to log a windows error message or some string received from the GUI which is in Utf-16 I would log it as a std::wstring and let the sink worry about it.